File: algorithm-problems/src/main/java/us/supercheng/algorithm/problems/leetcode/designparkingsystem/ParkingSystem.js
/**
 * @param {number} big
 * @param {number} medium
 * @param {number} small
 */
var ParkingSystem = function(big, medium, small) {
    this.map = {
        1: big,
        2: medium,
        3: small
    }
};

/**
 * @param {number} carType
 * @return {boolean}
 */
ParkingSystem.prototype.addCar = function(carType) {
    if (this.map[carType] > 0) {
        this.map[carType] -= 1;
        return true;
    }

    return false;
};

File: algorithm-problems/src/main/java/us/supercheng/algorithm/problems/leetcode/designparkingsystem/ParkingSystem.java
package us.supercheng.algorithm.problems.leetcode.designparkingsystem;

public class ParkingSystem {

    private int[] map;

    public ParkingSystem(int big, int medium, int small) {
        this.map = new int[4];
        this.map[1] = big;
        this.map[2] = medium;
        this.map[3] = small;
    }

    public boolean addCar(int carType) {
        if (this.map[carType] > 0) {
            this.map[carType]--;
            return true;
        }
        return false;
    }

}
